4-projects/modules/single_project/main.tf

```diff
@@ -21,7 +21,7 @@ module "project" {
   random_project_id           = "true"
   impersonate_service_account = var.impersonate_service_account
   activate_apis               = var.activate_apis
-  name                        = var.project_prefix
+  name                        = "prj-${var.business_code}-${element(split("", var.environment), 0)}-${var.project_prefix}"
```

Comment:
Instead of using `${element(split("", var.environment), 0)}` to get the env_code here and in other places in the code, we could use the declared variable env_code `${var.env_code}` or remove the variable `env_code` from the `variables.tf` file if it is not needed any more.

```
variable "env_code" {
  description = "A short form of the environment field"
  type        = string
}
```

Reply:
This should be resolved in latest commit, I think if we can infer that from environment no need to make users add two variables and potentially forget to update one.